c/misra/src/rules/RULE-8-5/ExternalObjectOrFunctionNotDeclaredInOneFile.ql

```diff
@@ -0,0 +1,43 @@
+/**
+ * @id c/misra/external-object-or-function-not-declared-in-one-file
+ * @name RULE-8-5: An external object or function shall be declared once in one and only one file
+ * @description Declarations in multiple files can lead to unexpected program behaviour.
+ * @kind problem
+ * @precision very-high
+ * @problem.severity warning
+ * @tags external/misra/id/rule-8-5
+ *       correctness
+ *       external/misra/obligation/required
+ */
+
+import cpp
+import codingstandards.c.misra
+
+from DeclarationEntry de, DeclarationEntry otherDeclaration, string kind
+where
+  not isExcluded(de, Declarations5Package::externalObjectOrFunctionNotDeclaredInOneFileQuery()) and
+  //this rule applies to non-defining declarations only
+  not de.isDefinition() and
+  not otherDeclaration.isDefinition() and
+  exists(Declaration d |
+    de.getDeclaration() = d and
+    otherDeclaration.getDeclaration() = d and
+    de.getFile() != otherDeclaration.getFile()
+  ) and
+  (
+    de.getDeclaration() instanceof Function and kind = "function"
+    or
+    de.getDeclaration() instanceof Variable and
+    not de.getDeclaration() instanceof Parameter and
+    kind = "variable"
+  ) and
+  // Apply an ordering based on location to enforce that (de1, de2) = (de2, de1) and we only report (de1, de2).
+  (
+    de.getFile().getAbsolutePath() < otherDeclaration.getFile().getAbsolutePath()
+    or
+    de.getFile().getAbsolutePath() = otherDeclaration.getFile().getAbsolutePath() and
```

Comment:
This check seems redundant to me since the query already restricts the results to
`de.getFile() != otherDeclaration.getFile()` on L25. If it is for some reason that I am missing necessary, why not `de.getFile() = otherDeclaration.getFile()`?

Reply:
you are completely correct, the ordering stuff here was something I copied but did not check to make sure that the second condition actually made sense (which it did not in this case). another good catch, thanks! :)